### PR TITLE
[subiquity_client] import types.dart generated from subiquity/types.py

### PR DESCRIPTION
**Pull request by jpnurmi**
Close: #412

### CODE REVIEW

File: packages/subiquity_client/lib/src/types.dart [hunk 1]
```diff
@@ -5,6 +5,36 @@ part 'types.g.dart';
 
 // ignore_for_file: invalid_annotation_target
 
+// subiquity serializes `Dict[int, Any]` as `List[List[int, Any]]`
+class JsonMapConverter
```
Comment (jpnurmi) [inline packages/subiquity_client/lib/src/types.dart:9]: I chose a more generator-friendly and property-agnostic syntax for the `Map<int, String>` JSON conversion workaround.

File: packages/subiquity_client/lib/src/types.dart [hunk 2]
```diff
@@ -5,6 +5,36 @@ part 'types.g.dart';
...
+  @override
+  List<dynamic> toJson(Map<int, String> values) {
+    return [
+      for (final entry in values.entries) [entry.key, entry.value],
+    ];
+  }
+}
+
+// subiquitycore.models.network
+class NetDevInfo {
```
Comment (jpnurmi) [inline packages/subiquity_client/lib/src/types.dart:29]: This is an unfortunate external dependency for `types.py`. This stub ensures that the generated code below won't choke on a missing type.

File: packages/subiquity_client/lib/src/types.dart [hunk 3]
```diff
@@ -5,6 +5,36 @@ part 'types.g.dart';
...
+  const NetDevInfo._();
+  factory NetDevInfo.fromJson(Map<String, dynamic> json) => NetDevInfo._();
+  Map<String, dynamic> toJson() => {};
+}
+
+// Union[Partition, Gap]
+typedef DiskObject = PartitionOrGap;
```
Comment (jpnurmi) [inline packages/subiquity_client/lib/src/types.dart:36]: the generator doesn't know that `Union[Partition, Gap] == DiskObject` so this is for backward compatibility

File: packages/subiquity_client/lib/src/types.dart [hunk 4]
```diff
@@ -71,23 +101,63 @@ class ApplicationStatus with _$ApplicationStatus {
...
+@freezed
+class KeyFingerprint with _$KeyFingerprint {
+  const factory KeyFingerprint({
+    required String keytype,
+    required String fingerprint,
+  }) = _KeyFingerprint;
+
+  factory KeyFingerprint.fromJson(Map<String, dynamic> json) =>
+      _$KeyFingerprintFromJson(json);
 }
```
Comment (jpnurmi) [inline packages/subiquity_client/lib/src/types.dart:119]: there are a few new types that aren't in use in the GUI today that codecov complains about. the total coverage goes only down by 0.7% so i think it's not a big deal :)

**Review by CarlosNihelton — APPROVED** [review]
This looks good to me. But it's still marked as draft. Do you plan on doing anything else?

**Comment by jpnurmi** [comment]
Sorry, my bad. :) I was feeling unsure if I dare to enable the generator check in the CI. It's ok now that it was left out.

**Pull request merged**